### PR TITLE
feat: add flutter replay docs

### LINKED ISSUE #11555: Add Documentation for Flutter Replay
### SDK

Flutter SDK

### Description

Experimental support for Flutter (mobile, not Web) Session Replay:
https://github.com/getsentry/sentry-dart/issues/1193#issuecomment-2403662575

### Suggested Solution

Copy [docs from React Native](https://docs.sentry.io/platforms/react-native/session-replay/) and adjust code snippets and any Flutter specific bits.

### PR DESCRIPTION
**Pull request by vaind**
## DESCRIBE YOUR PR

Resolves #11555

additionally a minor fix to all other mobile replay docs (copy-pasta)

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## LEGAL BOILERPLATE

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: docs/platforms/flutter/session-replay/index.mdx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+</Note>
+
+[Session Replay](/product/explore/session-replay/) helps you get to the root cause of an error or latency issue faster by providing you with a reproduction of what was happening in the user's device before, during, and after the issue. You can rewind and replay your application's state and see key user interactions, like taps, swipes, network requests, and console entries, in a single UI.
+
+By default, our Session Replay SDK masks all text content, images, and user input, giving you heightened confidence that no sensitive data will leave the device. To learn more, see [product docs](/product/explore/session-replay/).
+
+## Pre-requisites
+
+Make sure your Sentry Flutter SDK version is at least 8.9.0, which is required for Session Replay. 
+You can update your `pubspec.yaml` to the matching version:
```
Comment (lizokm) [inline docs/platforms/flutter/session-replay/index.mdx:23]:
```suggestion
## Prerequisites

Sentry Flutter SDK version 8.9.0 and up. You can update your `pubspec.yaml` by running the below:
```

File: docs/platforms/flutter/session-replay/index.mdx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+```yaml
+dependencies:
+  sentry_flutter: ^8.9.0
+```
+
+## Set Up
```
Comment (lizokm) [inline docs/platforms/flutter/session-replay/index.mdx:30]:
```suggestion
## Setup
```

File: docs/platforms/flutter/session-replay/index.mdx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+To set up the integration, add the following to your Sentry initialization.
```
Comment (lizokm) [inline docs/platforms/flutter/session-replay/index.mdx:32]:
```suggestion
To set up the integration, add the following to your Sentry initialization:
```

File: docs/platforms/flutter/session-replay/index.mdx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+While you're testing, we recommend that you set <PlatformIdentifier name="replays-session-sample-rate" /> to `1.0`. This ensures that every user session will be sent to Sentry.
+
+Once testing is complete, **we recommend lowering this value in production**. We still recommend keeping <PlatformIdentifier name="replays-on-error-sample-rate" /> set to `1.0`.
+
+## Sampling
+
+Sampling allows you to control how much of your website's traffic will result in a Session Replay. There are two sample rates you can adjust to get the replays relevant to you:
+
+1. <PlatformIdentifier name="replays-session-sample-rate" /> - The sample rate for
+   replays that begin recording immediately and last the entirety of the user's session.
```
Comment (lizokm) [inline docs/platforms/flutter/session-replay/index.mdx:56]:
```suggestion
1. <PlatformIdentifier name="replays-session-sample-rate" /> - The sample rate for
   replays that begin recording immediately and last the entirety of a user's session.
```

File: docs/platforms/flutter/session-replay/index.mdx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+2. <PlatformIdentifier name="replays-on-error-sample-rate" /> - The sample rate for
+   replays that are recorded when an error happens. This type of replay will record
+   up to a minute of events prior to the error and continue recording until the session
+   ends.
+
+Sampling begins as soon as a session starts. <PlatformIdentifier name="replays-session-sample-rate" /> is evaluated first. If it's sampled, the replay recording will begin. Otherwise, <PlatformIdentifier name="replays-on-error-sample-rate" /> is evaluated and if it's sampled, the integration will begin buffering the replay and will only upload it to Sentry if an error occurs. The remainder of the replay will behave similarly to a whole-session replay.
+
```
Comment (lizokm) [inline docs/platforms/flutter/session-replay/index.mdx:63]:
```suggestion
Sampling starts as soon as a session begins. The <PlatformIdentifier name="replays-session-sample-rate" /> is then evaluated. If the session is sampled, replay recording will start immediately. If not, <PlatformIdentifier name="replays-on-error-sample-rate" /> will be evaluated. If the session is sampled at this point, the replay will be buffered and will only be uploaded to Sentry if an error occurs. 

```

File: docs/platforms/flutter/session-replay/index.mdx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  options.experimental.replay.mask<IconButton>();
+  options.experimental.replay.unmask<Image>();
+  options.experimental.replay.maskCallback<Text>(
+      (Element element, Text widget) =>
+          (widget.data?.contains('secret') ?? false)
+              ? SentryMaskingDecision.mask
+              : SentryMaskingDecision.continueProcessing);
+```
+
+You can find more details in respective methods' code documentation.
```
Comment (lizokm) [inline docs/platforms/flutter/session-replay/index.mdx:80]:
```suggestion
You can find more details in the documentation for each method.
```

File: docs/platforms/flutter/session-replay/index.mdx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+<Note>
+
+If you encounter any data not being redacted with the default settings, please let us know through a [GitHub issue](https://github.com/getsentry/sentry-dart/issues/new?template=BUG_REPORT.yml).
+
```
Comment (lizokm) [inline docs/platforms/flutter/session-replay/index.mdx:85]:
```suggestion
If you find that data isn't being redacted with the default settings, please let us know by creating a [GitHub issue](https://github.com/getsentry/sentry-dart/issues/new?template=BUG_REPORT.yml).

```

File: docs/product/explore/session-replay/mobile/index.mdx [hunk 8]
```diff
@@ -56,13 +56,14 @@ These breadcrumbs are synced with the replay player and will auto-scroll as the
 
 ## SDKs Supported
 
-Session Replay for mobile is currently available for Android, iOS, and React Native SDKs. Support for [Flutter](/platforms/flutter/) is being tracked on this [GitHub issue](https://github.com/getsentry/sentry-dart/issues/1193).
+Session Replay for mobile is currently available for Android and iOS on both native SDKs as well as React Native and Flutter.
```
Comment (lizokm) [inline docs/product/explore/session-replay/mobile/index.mdx:59]:
```suggestion
Session Replay for mobile is currently available for Android and iOS on both native SDKs, as well as for React Native and Flutter.
```

File: docs/product/explore/session-replay/mobile/index.mdx [hunk 9]
```diff
@@ -56,13 +56,14 @@ These breadcrumbs are synced with the replay player and will auto-scroll as the
...
 
 We recommend updating to the latest, but the miminum versions supported are:
```
Comment (lizokm) [inline docs/product/explore/session-replay/mobile/index.mdx:61]:
```suggestion
We recommend updating to the latest version, but the minimum versions supported are:
```

**Review by lizokm — APPROVED** [review]
Made some wording suggestions.

File: docs/platforms/flutter/session-replay/index.mdx (hunk 1 above)
Comment (vaind) [reply docs/platforms/flutter/session-replay/index.mdx:23]: > by running the below:

that doesn't work here as below is not an executable script. It's a snippet of the configuration file and they already would have the `sentry_flutter` dependency added - they just need to make sure they have the minimum version with replay available.

File: docs/platforms/flutter/session-replay/index.mdx (hunk 5 above)
Comment (vaind) [reply docs/platforms/flutter/session-replay/index.mdx:63]: FYI same code is in other platform's session-replay/index.mdx files

**Comment by vaind** [comment]
> Made some wording suggestions.

Many thanks!

**Pull request merged**